Makefile

```
@@ -36,85 +50,81 @@ LDFLAGS += -march=rv32i_zicsr_zifencei -nostartfiles \
-		$(RISCV_PREFIX)-gcc -c $(CFLAGS) -Iut/riscv-tests/env/p -Iut/riscv-tests/isa/macros/scalar -Iut/riscv-tests/isa/rv32ui -o tests-build/$$test.o tests/$$test.S; \
-		$(RISCV_PREFIX)-gcc -o tests-build/$$test.elf $(LDFLAGS) tests-build/$$test.o sw/mtvec_handler.o; \
-		$(RISCV_PREFIX)-objcopy -j .text -j .data -j .rodata -O verilog --verilog-data-width=4 tests-build/$$test.elf tests-build/$$test.mem; \
+	for testfile in $(wildcard $</*.S) ; do \
+		test=$${testfile##*/}; test=$${test%.*}; \
+		$(RISCV_PREFIX)-gcc -c $(CFLAGS) -Iut/riscv-tests/env/p -Iut/riscv-tests/isa/macros/scalar -Iut/riscv-tests/isa/rv32ui -o $</$$test.o $</$$test.S; \
+		$(RISCV_PREFIX)-gcc -o $</$$test.elf $(LDFLAGS) $</$$test.o sw/mtvec_handler.o; \
+		$(RISCV_PREFIX)-objcopy -j .text -j .data -j .rodata -O verilog --verilog-data-width=4 $</$$test.elf $</$$test.mem; \
 	done
 
-run_vivado: compile_tests
+test: compile_tests
```

checkmake (0.3.2)

[warning] Target body for "test" exceeds allowed length of 5 lines (33).

(maxbodylength)